Support libxc 7.0.0

Trying to regression build 2024.3 rel against libxc 7.0.0, but run into some build issue. The error log is as below:


<details>
<summary>error build log</summary>

```
  Undefined symbols for architecture arm64:
    "___xc_f03_lib_m_MOD_xc_f03_func_end", referenced from:
        ___xc_libxc_MOD_libxc_lsd_eval in xc_libxc.F.o
        ___xc_libxc_MOD_libxc_lda_eval in xc_libxc.F.o
        ___xc_libxc_MOD_libxc_lsd_info in xc_libxc.F.o
        ___xc_libxc_MOD_libxc_lda_info in xc_libxc.F.o
        ___xc_libxc_MOD_libxc_add_sections in xc_libxc.F.o
        ___xc_libxc_MOD_libxc_get_reference_length in xc_libxc.F.o
        ___xc_libxc_wrap_MOD_xc_libxc_wrap_needs_laplace in xc_libxc_wrap.F.o
        ...
    "___xc_f03_lib_m_MOD_xc_f03_func_get_info", referenced from:
        ___xc_libxc_MOD_libxc_lsd_eval in xc_libxc.F.o
        ___xc_libxc_MOD_libxc_lda_eval in xc_libxc.F.o
        ___xc_libxc_MOD_libxc_lsd_info in xc_libxc.F.o
        ___xc_libxc_MOD_libxc_lda_info in xc_libxc.F.o
        ___xc_libxc_MOD_libxc_add_sections in xc_libxc.F.o
        ___xc_libxc_MOD_libxc_get_reference_length in xc_libxc.F.o
        ___xc_libxc_wrap_MOD_xc_libxc_wrap_needs_laplace in xc_libxc_wrap.F.o
        ...
    "___xc_f03_lib_m_MOD_xc_f03_func_info_get_ext_params_default_value", referenced from:
        ___xc_libxc_MOD_libxc_add_sections in xc_libxc.F.o
    "___xc_f03_lib_m_MOD_xc_f03_func_info_get_ext_params_description", referenced from:
        ___xc_libxc_MOD_libxc_add_sections in xc_libxc.F.o
        ___xc_libxc_wrap_MOD_xc_libxc_wrap_info_refs in xc_libxc_wrap.F.o
        ___xc_libxc_wrap_MOD_xc_libxc_get_reference_length in xc_libxc_wrap.F.o
        ___xc_libxc_wrap_MOD_xc_libxc_get_reference_length in xc_libxc_wrap.F.o
        ___xc_libxc_wrap_MOD_xc_libxc_get_reference_length in xc_libxc_wrap.F.o
        ___xc_libxc_wrap_MOD_xc_libxc_get_reference_length in xc_libxc_wrap.F.o
    "___xc_f03_lib_m_MOD_xc_f03_func_info_get_ext_params_name", referenced from:
        ___xc_libxc_MOD_libxc_add_sections in xc_libxc.F.o
        ___xc_libxc_wrap_MOD_xc_libxc_wrap_functional_set_params in xc_libxc_wrap.F.o
    "___xc_f03_lib_m_MOD_xc_f03_func_info_get_family", referenced from:
        ___xc_libxc_MOD_libxc_lsd_calc.constprop.0.isra.0 in xc_libxc.F.o
        ___xc_libxc_MOD_libxc_lda_eval._omp_fn.0 in xc_libxc.F.o
        ___xc_libxc_MOD_libxc_lsd_eval in xc_libxc.F.o
        ___xc_libxc_MOD_libxc_lsd_eval in xc_libxc.F.o
        ___xc_libxc_MOD_libxc_lsd_eval in xc_libxc.F.o
        ___xc_libxc_MOD_libxc_lda_eval in xc_libxc.F.o
        ___xc_libxc_MOD_libxc_lda_eval in xc_libxc.F.o
        ___xc_libxc_MOD_libxc_lda_eval in xc_libxc.F.o
        ...
    "___xc_f03_lib_m_MOD_xc_f03_func_info_get_flags", referenced from:
        ___xc_libxc_wrap_MOD_xc_libxc_wrap_functional_set_params in xc_libxc_wrap.F.o
        ___xc_libxc_wrap_MOD_xc_libxc_wrap_functional_set_params in xc_libxc_wrap.F.o
        ___xc_libxc_wrap_MOD_xc_libxc_wrap_needs_laplace in xc_libxc_wrap.F.o
        ___xc_libxc_wrap_MOD_xc_libxc_wrap_is_under_development in xc_libxc_wrap.F.o
    "___xc_f03_lib_m_MOD_xc_f03_func_info_get_kind", referenced from:
        ___xc_libxc_MOD_libxc_lsd_info in xc_libxc.F.o
        ___xc_libxc_MOD_libxc_lda_info in xc_libxc.F.o
    "___xc_f03_lib_m_MOD_xc_f03_func_info_get_n_ext_params", referenced from:
        ___xc_libxc_MOD_libxc_add_sections in xc_libxc.F.o
        ___xc_libxc_wrap_MOD_xc_libxc_wrap_functional_set_params in xc_libxc_wrap.F.o
        ___xc_libxc_wrap_MOD_xc_libxc_wrap_info_refs in xc_libxc_wrap.F.o
        ___xc_libxc_wrap_MOD_xc_libxc_get_reference_length in xc_libxc_wrap.F.o
        ___xc_libxc_MOD_libxc_lsd_calc.constprop.0.isra.0 in xc_libxc.F.o
        ___xc_libxc_MOD_libxc_lsd_calc.constprop.0.isra.0 in xc_libxc.F.o
        ...
    "___xc_f03_lib_m_MOD_xc_f03_mgga_exc", referenced from:
        ___xc_libxc_MOD_libxc_lsd_calc.constprop.0.isra.0 in xc_libxc.F.o
        ___xc_libxc_MOD_libxc_lsd_calc.constprop.0.isra.0 in xc_libxc.F.o
        ___xc_libxc_MOD_libxc_lsd_calc.constprop.0.isra.0 in xc_libxc.F.o
        ___xc_libxc_MOD_libxc_lda_eval._omp_fn.0 in xc_libxc.F.o
        ___xc_libxc_MOD_libxc_lda_eval._omp_fn.0 in xc_libxc.F.o
        ___xc_libxc_MOD_libxc_lda_eval._omp_fn.0 in xc_libxc.F.o
        ___xc_libxc_MOD_libxc_lda_eval._omp_fn.0 in xc_libxc.F.o
        ___xc_libxc_MOD_libxc_lda_eval._omp_fn.0 in xc_libxc.F.o
        ___xc_libxc_MOD_libxc_lda_eval._omp_fn.0 in xc_libxc.F.o
        ___xc_libxc_MOD_libxc_lda_eval._omp_fn.0 in xc_libxc.F.o
        ...
    "___xc_f03_lib_m_MOD_xc_f03_mgga_exc_vxc", referenced from:
        ___xc_libxc_MOD_libxc_lsd_calc.constprop.0.isra.0 in xc_libxc.F.o
        ___xc_libxc_MOD_libxc_lda_eval._omp_fn.0 in xc_libxc.F.o
    "___xc_f03_lib_m_MOD_xc_f03_mgga_exc_vxc_fxc", referenced from:
        ___xc_libxc_MOD_libxc_lsd_calc.constprop.0.isra.0 in xc_libxc.F.o
        ___xc_libxc_MOD_libxc_lsd_calc.constprop.0.isra.0 in xc_libxc.F.o
        ___xc_libxc_MOD_libxc_lda_eval._omp_fn.0 in xc_libxc.F.o
        ___xc_libxc_MOD_libxc_lda_eval._omp_fn.0 in xc_libxc.F.o
    "___xc_f03_lib_m_MOD_xc_f03_mgga_vxc", referenced from:
        ___xc_libxc_MOD_libxc_lsd_calc.constprop.0.isra.0 in xc_libxc.F.o
        ___xc_libxc_MOD_libxc_lsd_calc.constprop.0.isra.0 in xc_libxc.F.o
        ___xc_libxc_MOD_libxc_lda_eval._omp_fn.0 in xc_libxc.F.o
        ___xc_libxc_MOD_libxc_lda_eval._omp_fn.0 in xc_libxc.F.o
        ___xc_libxc_MOD_libxc_lda_eval._omp_fn.0 in xc_libxc.F.o
        ___xc_libxc_MOD_libxc_lda_eval._omp_fn.0 in xc_libxc.F.o
    "___xc_f03_lib_m_MOD_xc_f03_mgga_vxc_fxc", referenced from:
        ___xc_libxc_MOD_libxc_lsd_calc.constprop.0.isra.0 in xc_libxc.F.o
        ___xc_libxc_MOD_libxc_lsd_calc.constprop.0.isra.0 in xc_libxc.F.o
        ___xc_libxc_MOD_libxc_lda_eval._omp_fn.0 in xc_libxc.F.o
        ___xc_libxc_MOD_libxc_lda_eval._omp_fn.0 in xc_libxc.F.o
  ld: symbol(s) not found for architecture arm64
  collect2: error: ld returned 1 exit status
```


</details>

full build log, https://github.com/Homebrew/homebrew-core/actions/runs/11337605254/job/31529529142?pr=194402
relates to Homebrew/homebrew-core#194402

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -613,7 +613,7 @@ if(CP2K_USE_ELPA)
 endif()
 
 if(CP2K_USE_LIBXC)
-  find_package(LibXC 6 REQUIRED EXACT)
+  find_package(LibXC 7 REQUIRED)
 endif()
 
 # uncomment this when libgrpp cmake support is complete

--- a/cmake/cp2kConfig.cmake.in
+++ b/cmake/cp2kConfig.cmake.in
@@ -60,7 +60,7 @@ if(NOT TARGET cp2k::cp2k)
   endif()
 
   if(@CP2K_USE_LIBXC@)
-    find_dependency(LibXC 6 REQUIRED EXACT)
+    find_dependency(LibXC 7 REQUIRED)
   endif()
 
   if(@CP2K_USE_COSMA@)

--- a/cmake/modules/FindLibXC.cmake
+++ b/cmake/modules/FindLibXC.cmake
@@ -13,11 +13,7 @@ cp2k_set_default_paths(LIBXC "LibXC")
 
 if(PKG_CONFIG_FOUND)
   # For LibXC >= 7, the Fortran interface is only libxcf03
-  pkg_check_modules(CP2K_LIBXC
-    IMPORTED_TARGET GLOBAL
-    libxcf03
-    libxc>=7
-  )
+  pkg_check_modules(CP2K_LIBXC IMPORTED_TARGET GLOBAL libxcf03 libxc>=7)
 endif()
 
 if(NOT CP2K_LIBXC_FOUND)
@@ -32,8 +28,7 @@ endif()
 if(CP2K_LIBXC_FOUND)
   # Require both libxc + libxcf03 for LibXC 7
   set(CP2K_LIBXC_LINK_LIBRARIES
-    "${CP2K_LIBXCF03_LIBRARIES};${CP2K_LIBXC_LIBRARIES}"
-  )
+      "${CP2K_LIBXCF03_LIBRARIES};${CP2K_LIBXC_LIBRARIES}")
 endif()
 
 if(NOT CP2K_LIBXC_INCLUDE_DIRS)

--- a/cmake/modules/FindLibXC.cmake
+++ b/cmake/modules/FindLibXC.cmake
@@ -12,8 +12,12 @@ include(cp2k_utils)
 cp2k_set_default_paths(LIBXC "LibXC")
 
 if(PKG_CONFIG_FOUND)
-  pkg_check_modules(CP2K_LIBXC IMPORTED_TARGET GLOBAL libxcf03
-                    libxc>=${LibXC_FIND_VERSION})
+  # For LibXC >= 7, the Fortran interface is only libxcf03
+  pkg_check_modules(CP2K_LIBXC
+    IMPORTED_TARGET GLOBAL
+    libxcf03
+    libxc>=7
+  )
 endif()
 
 if(NOT CP2K_LIBXC_FOUND)
@@ -25,9 +29,11 @@ if(NOT CP2K_LIBXC_FOUND)
   endforeach()
 endif()
 
-if(CP2K_LIBXC_FOUND AND CP2K_LIBXCF03_FOUND)
+if(CP2K_LIBXC_FOUND)
+  # Require both libxc + libxcf03 for LibXC 7
   set(CP2K_LIBXC_LINK_LIBRARIES
-      "${CP2K_LIBXCF03_LIBRARIES};${CP2K_LIBXC_LIBRARIES}")
+    "${CP2K_LIBXCF03_LIBRARIES};${CP2K_LIBXC_LIBRARIES}"
+  )
 endif()
 
 if(NOT CP2K_LIBXC_INCLUDE_DIRS)

